### PR TITLE
digitalocean: Add support for DigitalOcean

### PR DESCRIPTION
**Pull request by aknuds1**
This extends Tectonic Installer to also support the DigitalOcean cloud
provider.

### CODE REVIEW

File: platforms/digitalocean/resources/do-pusher.sh [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+export ACCESS_KEY_ID=${do_access_key_id}
+export SECRET_ACCESS_KEY=${do_secret_access_key}
+export REGION=${do_region}
+
+# shellcheck disable=SC2034,SC1083
+filename=$(basename $${1})
+# shellcheck disable=SC2034,SC1083
+cp $${1} /tmp
+# A tool for interacting with DO's object storage: https://github.com/aknuds1/do-spaces-tool
+docker pull aknudsen/do-spaces-tool:0.2.0 > /dev/null
```
Comment (enxebre) [inline platforms/digitalocean/resources/do-pusher.sh]: we'll need to mirror this into quay.io/coreos

**Comment by aknuds1** [comment]
@enxebre @sym3tri I just rebased on track-1. I've tested that the Terraform plan compiles through `terraform plan`, but not actually created any cluster yet.

File: platforms/digitalocean/resources/do-pusher.sh (hunk 1 above)
Comment (sym3tri) [reply platforms/digitalocean/resources/do-pusher.sh]: We can do that ourselves in a follow up.

**Comment by sym3tri** [comment]
@aknuds1 sure, we've got it. I'll ping you if we see any major failures, but I think it'll be fine. Thanks again for all your hard work on this.

**Comment by aknuds1** [comment]
@sym3tri Thanks Ed, really looking forward to seeing this merge :)

**Comment by kalmog** [comment]
ok to test

**Comment by kalmog** [comment]
test this please

**Comment by kalmog** [comment]
aws is green. re-running azure + metal

**Comment by kalmog** [comment]
retest this please

**Comment by kalmog** [comment]
metal: retest this please

**Comment by kalmog** [comment]
retest this please (metal)

**Comment by kalmog** [comment]
retest this please

**Comment by kalmog** [comment]
this PR is cherry-picked to the 1.8.7 branch, in #2948

**Comment by aknuds1** [comment]
@kalmog Finally! Fantastic!

**Pull request closed without merging**